internal/controller/promotions/promotions.go

```diff
@@ -106,16 +118,20 @@ func SetupReconcilerWithManager(
 }
```

Comment:
Nit: I see that the task of building up the `reconcilerName` is accomplished a bit differently here and in the Stage reconciler. I am pretty sure that discrepancy is because the Stage reconciler has a `ReconcilerConfig` type that helped you deal with things in a nicer way and no such type exists here for the Promotion reconciler...

I like the way it was done for the Stage reconciler better, and even if it means introducing a new `ReconcilerConfig` type here for the Promotions reconciler, I'd be in favor of that.